File: app/src/main/java/com/diu/yk_games/line2box/DataStore.java
package com.diu.yk_games.line2box;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.util.Log;

public class DataStore
{
    public String timeData="";
    public String redData="Red";
    public String blueData="Blue";
    public String starData="";
    public String plr1Id="";
    public String plr2Id="";
    public String plr1Cup="";
    public String plr2Cup="";
    public DataStore(){}


    public DataStore(String timeData, String redData, String blueData, String starData, String plr1Id, String plr2Id, String plr1Cup, String plr2Cup) {
        this.timeData = timeData;
        this.redData = redData;
        this.blueData = blueData;
        this.starData = starData;
        this.plr1Id = plr1Id;
        this.plr2Id = plr2Id;
        this.plr1Cup = plr1Cup;
        this.plr2Cup = plr2Cup;
    }
}

File: app/src/main/java/com/diu/yk_games/line2box/model/DataStore.kt
package com.diu.yk_games.line2box.model

import android.os.Parcelable
import com.google.firebase.firestore.IgnoreExtraProperties
import kotlinx.parcelize.Parcelize

@Parcelize
@IgnoreExtraProperties
data class DataStore(
    var time: Long = 0L,
    var redData: String = "Red",
    var blueData: String = "Blue",
    var starData: String = "",
    var plr1Id: String = "",
    var plr2Id: String = "",
    var plr1Cup: String = "0",
    var plr2Cup: String = "0"
) : Parcelable {
    fun toScore(): Score? {
        val red = redData.split(": ")
        val blue = blueData.split(": ")
        if (red.size != 2 || blue.size != 2) return null
        return Score(
            time = time,
            type = if (starData == "friendly") Score.Type.Friendly else Score.Type.Globe,
            player1 = PlayerInfo(id = plr1Id, nm = red.first()),
            player2 = PlayerInfo(id = plr2Id, nm = blue.first()),
            result = LiveResult(
                score1 = red.last().toInt(),
                score2 = blue.last().toInt(),
                cup1 = plr1Cup,
                cup2 = plr2Cup
            )
        )
    }
}